airflow/providers/dbt/cloud/utils/openlineage.py

```diff
@@ -21,13 +21,25 @@
 from contextlib import suppress
 from typing import TYPE_CHECKING
 
+from airflow import __version__ as airflow_version
+
 if TYPE_CHECKING:
     from airflow.models.taskinstance import TaskInstance
     from airflow.providers.dbt.cloud.operators.dbt import DbtCloudRunJobOperator
     from airflow.providers.dbt.cloud.sensors.dbt import DbtCloudJobRunSensor
     from airflow.providers.openlineage.extractors.base import OperatorLineage
 
 
+def _get_try_number(val):
+    # todo: remove when min airflow version >= 2.10.0
+    from packaging.version import parse
+
+    if parse(parse(airflow_version).base_version) < parse("2.10.0"):
+        return val.try_number - 1
+    else:
+        return val.try_number
```

Comment:
I don't love this, but other than deleting the `_try_number` property (and using that to detect the older version, possibly breaking back compat) I don't have any other ideas.

Reply:
SHould we maybe raise deprecation warnings if this is used in the code somewhere?

Reply:
That's what this is currently doing, `ti._try_number` is currently raising the deprecation warning. But that means we can't use `if hasattr(val, "_try_number")` as a way of detecting the old version, cos we've done the "right" thing.

Reply:
yeah.  @ashb originally i was just deleting _try_number.  but the reason we added that setter is just elasticsearch task handler.  there's one code path where it will set `_try_number`  since it's in providers, there will be users who upgrade airflow without upgrading the provider, and they will hit it. ideally should not have been referencing _try_number in the first place.  (my mistake actually). so this is just going out of the way to be nice to such users.

re this ugly compat shim.... we could probably still check by looking at the object... but i think @jedcunningham's argument was that comparing version is more explicit / self-documenting, and i think i agree.

Reply:
e.g. when _try_number is a property then Class._try_number will be a property object
and when _try_number is an instance attribute then Class._try_number will not be present
but yeah... addressed why looking at version seems better above